pkg/disk/disk.go

```diff
@@ -32,3 +46,87 @@ type Info struct {
 	// Usage is calculated per tenant.
 	Usage uint64
 }
+
+// Performance holds informantion about read and write speed of a disk
+type Performance struct {
+	Path       string  `json:"path"`
+	Error      string  `json:"error,omitempty"`
+	WriteSpeed float64 `json:"writeSpeed"`
+	ReadSpeed  float64 `json:"readSpeed"`
+}
+
+// GetPerformance returns given disk's read and write performance
+func GetPerformance(path string) Performance {
+	perf := Performance{}
+	write, read, err := performance(path)
+	if err != nil {
+		perf.Error = err.Error()
+		return perf
+	}
+	perf.WriteSpeed = write
+	perf.ReadSpeed = read
+	return perf
+}
+
+// Calculate the write and read performance - write and read 10 tmp (1 MiB)
+// files and find the average time taken (Bytes / Sec)
+func performance(fsPath string) (write, read float64, err error) {
+	var count int
+	var totalWriteElapsed time.Duration
+	var totalReadElapsed time.Duration
+	buf := make([]byte, fileSize)
+
+	// create the enclosing directory
+	err = os.Mkdir(fsPath, 0777)
+	if err != nil {
+		cleanup(fsPath)
+		return 0, 0, err
+	}
+
+	for count = 1; count <= 10; count++ {
+		rand.Read(buf)
```

Comment:
We can avoid doing rand.Read() since it is taxing CPU you can simply use bytes.Repeat perhaps or write just zeros for this.

Reply:
use rand.Read() once for 1 KiB and then using bytes.Repeat